service/project/CMakeLists.txt

```diff
@@ -67,6 +69,8 @@ add_jar(corethriftjava
   ${CMAKE_CURRENT_SOURCE_DIR}/gen-java/cc/service/core/Timeout.java
   OUTPUT_NAME corethrift)
 
+add_dependencies(corethriftjava commonthrift)
```

Comment:
I found an easier solution. This will make the Java compilation wait until the C++ compilation of the Thrift types is done, but is *much easier* to express in the Makefile.